src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,73 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SEPARATE_COMMA = ",";
+    private static final int AMOUNT_INDEX = 1;
+    private static final int OPERATION_INDEX = 0;
+    private static final int RESULT_VALUE = 0;
+    private static final String BUY_VALUE = "buy";
+    private static final String SUPPLY_VALUE = "supply";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String value = readFile(fromFileName);
+        String finalReport = createReport(value);
+        writeFile(toFileName, finalReport);
+    }
+
+    private String readFile(String fromFileName) {
+        File file = new File(fromFileName);
+        try {
+            BufferedReader reader = new BufferedReader(new FileReader(file));
```

Comment:
It is recommended to use try-with-resources statement to ensure that the BufferedReader is closed properly.

Reply:
OK fix that